### DISCUSSION TITLE
Is it possible to distribute the required amount of memory with Mishard?

### DISCUSSION THREAD

**Discussion by rky0930** [milvus-io/milvus#4899] [Q&A and General discussion] [answered]
Is it possible to distribute the required amount of memory with Mishard?

Since only the necessary segments are loaded into the memory of each milvus-readonly server, it seems to have an effect of memory distribution. Is that right ?

If, according to the size-tool calculation, a total of 30Gb of memory is required, can all milvus-read-only servers have 30Gb of memory to keep the cache? Or just only need 10Gb each?

**Comment by yhmo** [comment, marked as answer]
Milvus store data in three levels: [collection, partition, segment](https://www.milvus.io/docs/v1.0.0/storage_concept.md). When search request is received, milvus load segments into memory and search each segment, then reduce result to final result.
Mishards use [consistent hashing ring](https://en.wikipedia.org/wiki/Consistent_hashing) to distribute segment data in nodes. For each segment, mishard get its id and hash to a value, then send the segment to a node according to hash ring. But typically segments are not averagely distributed to each node. For example, you have 30Gb data, 30 segments, 3 nodes. Maybe the node_1 get 11 segments, node_2 get 9 segments, node_3 get 10 segments.
